rest-service/manager_rest/cluster_status_manager.py

```diff
@@ -303,6 +308,59 @@ def _get_db_cluster_status(db_service, expected_nodes_number):
     return db_service[STATUS]
 
 
+def _should_not_validate_cluster_status(service):
+    return service[STATUS] == ServiceStatus.FAIL or service[IS_EXTERNAL]
+
+
+def _get_broker_cluster_status(broker_service, expected_nodes_number):
+    if _should_not_validate_cluster_status(broker_service):
+        return broker_service[STATUS]
+
+    broker_nodes = broker_service['nodes']
+    active_broker_nodes = {name: node for name, node in broker_nodes.items()
+                           if node[STATUS] == ServiceStatus.HEALTHY}
+
+    if not _verify_identical_broker_cluster_status(active_broker_nodes):
+        return ServiceStatus.FAIL
+
+    if expected_nodes_number != len(active_broker_nodes):
+        # This should happen only on broker setup
+        current_app.logger.error(
+            'There are {0} active broker nodes, but there are {1} broker nodes'
+            ' registered in the DB.'.format(active_broker_nodes,
+                                            expected_nodes_number))
+        return ServiceStatus.DEGRADED
+
+    return broker_service[STATUS]
+
+
+def _extract_broker_cluster_status_from_node(broker_node):
+    broker_service = broker_node[SERVICES][BROKER_SERVICE_KEY]
+    return broker_service[EXTRA_INFO]['cluster_status']
+
+
+def _compare_broker_cluster_statuses(node_a, node_b):
+    node_name_a, node_name_b = node_a[0], node_b[0]
+    cluster_status_a = _extract_broker_cluster_status_from_node(node_a[1])
+    cluster_status_b = _extract_broker_cluster_status_from_node(node_b[1])
+    if cluster_status_a != cluster_status_b:
+        current_app.logger.error(BROKER_NODES_DIFFERENT_CLUSTER_STATUS.
+                                 format(node_name_a=node_name_a,
+                                        cluster_status_a=cluster_status_a,
+                                        node_name_b=node_name_b,
+                                        cluster_status_b=cluster_status_b))
+        return False
+    return True
+
+
+def _verify_identical_broker_cluster_status(active_nodes):
```

Comment:
well this is a bit overkill, you just need one source of truth and compare to it